### PR TITLE
Error Functions

### PR DESCRIPTION
**Pull request by bwormser**
Added value error checking functions based on the fields described in the DB proposal.

### CODE REVIEW

File: client/src/helpers/errors.ts [hunk 1]
```diff
@@ -1,2 +1,69 @@
...
-export {}
\ No newline at end of file
+//Retruns true if passed a non empty string
+function checkNonEmptyString(str: string){
+    return (str.trim() !== '');
+}
+
+//Returns true if every entry of an array is a non empty string
+function checkArrayOfStrings(strs: Array<string>){
+    let retbool = true;
```
Comment (roccopolimen) [inline client/src/helpers/errors.ts]: unnecesary, remove this var

File: server/helpers/error.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    return true;
+}
+//Retruns true if passed a non empty string
+function checkNonEmptyString(str: string){
+    return (str.trim() !== '');
+}
+
+//Returns true if every entry of an array is a non empty string
+function checkArrayOfStrings(strs: Array<string>){
+    let retbool = true;
```
Comment (roccopolimen) [inline server/helpers/error.ts]: unnecesary var

**Review by roccopolimen — COMMENTED** [review]
Few comments, plus what I said in slack. 

Also do not submit `package-lock.json` file so re-commit code with it in the gitignore and removed from tracking along with the few comments and then I approve. Should take 5 minutes to fix everything for now.

File: client/src/helpers/errors.ts [hunk 3]
```diff
@@ -1,2 +1,69 @@
...
+    for(let str in strs) {
```
Comment (mpolimen) [inline client/src/helpers/errors.ts]: The "in" keyword enumerates for objects, while "of" is for arrays.
https://stackoverflow.com/questions/29285897/what-is-the-difference-between-for-in-and-for-of-statements

**Review by mpolimen — CHANGES_REQUESTED** [review]
Also delete package-lock.json

File: client/src/helpers/errors.ts (hunk 3 above)
Comment (bwormser) [reply client/src/helpers/errors.ts]: Okay I fixed this

File: client/src/helpers/errors.ts (hunk 1 above)
Comment (bwormser) [reply client/src/helpers/errors.ts]: Yep I got rid of it

**Pull request merged**